### PR TITLE
Parse and display video meta data

### LINKED ISSUE #19: Show Video title, description and license

### PR DESCRIPTION
**Pull request by timonegk**
Closes #19.

### CODE REVIEW

**Comment by JanGut** [comment]
TODOs:

- [ ] Update requirements in Readme
- [ ] Add MetaDataProvider for video files
- [ ] More beautiful css

File: lecture2gether_flask/meta_data_provider.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+
+class YouTubeMetaDataProvider(MetaDataProvider):
+    def __init__(self, video_url):
+        super().__init__(video_url)
+        self._video_id = youtube_video_id_from_url(video_url)
+
+    def get_meta_data(self):
+        youtube = googleapiclient.discovery.build(
+            'youtube', 'v3', developerKey=os.environ['GOOGLE_API_KEY'], cache_discovery=False)
```
Comment (Flova) [inline lecture2gether_flask/meta_data_provider.py:106]: Impl case if env variable is not set, aka. there is no API Key

File: README.md [hunk 2]
```diff
@@ -38,7 +38,10 @@ The video streams are synchronized to partially restore the social aspect of cam
 * [Flask-RESTful](https://flask-restful.readthedocs.io)
 * [Prometheus Flask exporter](https://github.com/rycus86/prometheus_flask_exporter)
 * [Prometheus Python Client](https://github.com/prometheus/client_python)
-* [Eventlet](http://eventlet.net)
+* [gevent](http://gevent.org)
+* [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/)
+* [Google API Python Client](https://github.com/googleapis/google-api-python-client/)
+* [Nose](https://nose.readthedocs.io/en/latest/)
 * [Redis](https://redis.io/)
 * [coolname](https://github.com/alexanderlukanin13/coolname)
```
Comment (Flova) [inline README.md:47]: Add env variable to the table below

**Comment by timonegk** [comment]
Currently, the video playing is completely broken... 😞

**Comment by timonegk** [comment]
For some reason, I had to store the video stream url separate from the meta data. I don't know why, but it is working now. @fredericobormann and others, please retest.

File: README.md (hunk 2 above)
Comment (timonegk) [reply README.md:47]: Done

File: lecture2gether_flask/test.py [hunk 3]
```diff
@@ -1,61 +1,155 @@
...
+
+def test_youtube_metadata():
+    if 'GOOGLE_API_KEY' not in os.environ:
+        return
+
+    meta_data_provider = YouTubeMetaDataProvider('https://youtu.be/qxyQCD3QT6Y')
+    meta_data = meta_data_provider.get_meta_data()
+    assert meta_data['url'] == 'https://www.youtube.com/watch?v=qxyQCD3QT6Y'
+    assert meta_data['streamUrl'] == 'https://www.youtube.com/watch?v=qxyQCD3QT6Y'
+    assert meta_data['title'] == 'Cravendale Last Marble Standing E2 Balancing - Marble Race by Jelle\'s Marble Runs'
```
Comment (JanGut) [inline lecture2gether_flask/test.py:100]: lol

File: lecture2gether_flask/app.py [hunk 4]
```diff
@@ -1,24 +1,26 @@
 #!/usr/bin/env python3
+from gevent import monkey
```
[the remark this answers is not in the document]
Comment (timonegk) [reply lecture2gether_flask/app.py:2]: Yes. https://www.gevent.org/api/gevent.monkey.html

File: lecture2gether_flask/app.py [hunk 5]
```diff
@@ -96,32 +98,29 @@ def decode_l2go_path():
     if not data or not 'video_url' in data:
         abort(400)
 
-    video_url = data['video_url']
+    _video_url = data['video_url']
```
[the remark this answers is not in the document]
Comment (JanGut) [reply lecture2gether_flask/app.py]: Dunno

**Pull request merged**